cmd/kueuectl/app/create/create_clusterqueue.go

```diff
@@ -0,0 +1,488 @@
+		"Defines which namespaces are allowed to submit workloads to this clusterQueue.")
+	cmd.Flags().StringVar(&o.UserSpecifiedReclaimWithinCohort, reclaimWithinCohort, "",
+		"Determines whether a pending Workload can preempt Workloads from other ClusterQueues in the cohort that are using more than their nominal quota.")
+	cmd.Flags().StringVar(&o.UserSpecifiedPreemptionWithinClusterQueue, preemptionWithinClusterQueue, "",
+		"Determines whether a pending Workload that doesn't fit within the nominal quota for its ClusterQueue, can preempt active Workloads in the ClusterQueue.")
+	cmd.Flags().StringSliceVar(&o.UserSpecifiedNominalQuota, nominalQuota, []string{},
+		"The quantity of this resource that is available for Workloads admitted by this ClusterQueue at a point in time.")
+	cmd.Flags().StringSliceVar(&o.UserSpecifiedBorrowingLimit, borrowingLimit, []string{},
+		"The maximum amount of quota for the [flavor, resource] combination that this ClusterQueue is allowed to borrow from the unused quota of other ClusterQueues in the same cohort.")
+	cmd.Flags().StringSliceVar(&o.UserSpecifiedLendingLimit, lendingLimit, []string{},
+		"The maximum amount of unused quota for the [flavor, resource] combination that this ClusterQueue can lend to other ClusterQueues in the same cohort.")
+
+	return cmd
+}
+
+// Complete completes all the required options
+func (o *ClusterQueueOptions) Complete(clientGetter util.ClientGetter, cmd *cobra.Command, args []string) error {
+	o.Name = args[0]
+
+	if cmd.Flags().Changed(queuingStrategy) {
+		o.QueueingStrategy = v1beta1.QueueingStrategy(o.UserSpecifiedQueueingStrategy)
+	}
+
+	if cmd.Flags().Changed(namespaceSelector) {
+		o.NamespaceSelector = metav1.LabelSelector{
+			MatchLabels: o.UserSpecifiedNamespaceSelector,
+		}
+	}
+
+	if cmd.Flags().Changed(reclaimWithinCohort) {
+		o.ReclaimWithinCohort = v1beta1.PreemptionPolicy(o.UserSpecifiedReclaimWithinCohort)
+	}
+
+	if cmd.Flags().Changed(preemptionWithinClusterQueue) {
+		o.PreemptionWithinClusterQueue = v1beta1.PreemptionPolicy(o.UserSpecifiedPreemptionWithinClusterQueue)
+	}
+
+	var err error
+	if cmd.Flags().Changed(nominalQuota) || cmd.Flags().Changed(borrowingLimit) || cmd.Flags().Changed(lendingLimit) {
+		err = o.parseResourceGroups()
+	}
+	if err != nil {
+		return err
+	}
+
+	clientset, err := clientGetter.KueueClientSet()
+	if err != nil {
+		return err
+	}
+
+	o.Client = clientset.KueueV1beta1()
+
+	o.DryRunStrategy, err = util.GetDryRunStrategy(cmd)
+	if err != nil {
+		return err
+	}
+
+	err = util.PrintFlagsWithDryRunStrategy(o.PrintFlags, o.DryRunStrategy)
+	if err != nil {
+		return err
+	}
+
+	printer, err := o.PrintFlags.ToPrinter()
+	if err != nil {
+		return err
+	}
+
+	o.PrintObj = printer.PrintObj
+
+	return nil
+}
+
+// Validate validates required fields are set to support structured generation
+func (o *ClusterQueueOptions) Validate(ctx context.Context) error {
```

Comment:
```suggestion
func (o *ClusterQueueOptions) validate(ctx context.Context) error {
```

Do we really need to expose this function?